lib/internal/modules/esm/default_resolve.js

```diff
@@ -43,11 +43,15 @@ const legacyExtensionFormatMap = {
   '.node': 'commonjs'
 };
 
-if (experimentalWasmModules)
+if (experimentalWasmModules) {
+  process.emitWarning('The WASM module is experimental.', 'ExperimentalWarning');
   extensionFormatMap['.wasm'] = legacyExtensionFormatMap['.wasm'] = 'wasm';
+}
 
-if (experimentalJsonModules)
+if (experimentalJsonModules) {
+  process.emitWarning('The JSON module is experimental.', 'ExperimentalWarning');
```

Comment:
The actual usage of these is for example at https://github.com/nodejs/node/blob/master/lib/internal/modules/esm/translators.js#L136.

I think moving the warning to the actual usage could be beneficial.